### ISSUE TITLE
Fix the remaining sphinx 1.4 warning

### ISSUE THREAD

**Issue by eteq** [astropy/astropy-helpers#227]
As discussed in #224, sphinx 1.4 has caused two new warnings to appear.  One was fixed in #224, but this one remains:
```
WARNING: while setting up extension astropy_helpers.sphinx.ext.autodoc_enhancements: directive 'autoattribute' is already registered, it will be overridden
```

And from https://github.com/astropy/astropy-helpers/pull/224#issuecomment-210034280:
>  The first warning is coming from this https://github.com/astropy/astropy-helpers/blob/master/astropy_helpers/sphinx/ext/autodoc_enhancements.py#L75
>
> I tried to catch the warning there and ignore it, but it didn't work. Getting the bug workaround into upstream sphinx may be the best long term solution.

**Comment by bsipocz** [comment]
btw @embray made a PR to sphinx with the patch here: https://github.com/sphinx-doc/sphinx/pull/1843

**Comment by eteq** [comment]
And I'm noticing you've suggested an in-sphinx fix in sphinx-doc/sphinx#2451

**Comment by bsipocz** [comment]
Yes, either way would work. I tried to implement the latter but over-complicated it, and still didn't manage to get it work.
